VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -291,7 +298,9 @@ public void onEventMainThread(DiscussionThreadPostedEvent event) {
             // move the ListView's scroll to that newly added post's position
             discussionPostsListView.setSelection(i);
             // In case this is the first addition, we need to hide the no-item-view
-            setNoResultView(null);
+            setScreenStateUponResult(null);
```

Comment:
It's a weird situation, because we only want to clear the error view when we have some results to display, and if the list view is empty then we want to display some error message, so in a sense the nullability of the parameter is predetermined for both cases. It might make sense to split this into two methods (e.g. `setScreenStateUponResult()` and setScreenStateUponError()`).